### PR TITLE
Enrich installation guide

### PR DESCRIPTION
**Pull request by shorim**
**Description**

Changes proposed in this pull request:

- add examples for using custom components and custom overrides when installing Kyma.
- more description for installing Kyma using custom domain.
- more description for installing Kyma from local sources to a remote cluster.
- change test data used for overrides, since current overrides are not applicable after `istio` upgrade to 1.6.
**Related issue(s)**

#610

### CODE REVIEW

File: docs/03-02-examples.md [hunk 1]
```diff
@@ -45,6 +45,42 @@ To build an image from your local sources and install Kyma based on this image,
...
+    labels:
+      installer: overrides
+      component: ory
+      kyma-project.io/installation: ""
+  data:
+    hydra.deployment.resources.requests.cpu: "53m"
+    hydra.deployment.resources.limits.cpu: "153m"
+  ```
+- In this example, the values of `hydra.deployment.resources.requests.cpu` and `hydra.deployment.resources.limits.cpu` will be overriden to `53m` and `153m` respectively.
+
```
Comment (a-thaler) [inline docs/03-02-examples.md:83]: Maybe add a remark that you can have multiple configmaps for different components in the same file and that you can repeat the "-o" flag with different files. Please use the full name of flags in docu like "--override"

File: docs/03-02-examples.md [hunk 2]
```diff
@@ -45,6 +45,42 @@ To build an image from your local sources and install Kyma based on this image,
 kyma install -s local --custom-image {IMAGE}
 ```
```
Comment (a-thaler) [inline docs/03-02-examples.md:47]: That is a veryy good starting point and is what we need. Please extend it more with other examples regarding the --source flag and custom domains

File: docs/03-02-examples.md [hunk 3]
```diff
@@ -26,24 +26,100 @@ To install Kyma using your own domain, run:
...
 
-To build an image from your local sources and install Kyma based on this image, run:
+To build an image from your local sources and install Kyma on a remote cluster based on this image, run:
 
 ```bash
-kyma install -s local --custom-image {IMAGE}
+kyma install --source local --custom-image {IMAGE}
 ```
+- For example, if your `{IMAGE}` is `user/my-kyma-installer:v1.4.0`, then this command will build an image from your local sources and push it your repository `user/my-kyma-installer` with the tag `v1.4.0`. Then, this image `user/my-kyma-installer:v1.4.0` will be used in the `Deployment` for `kyma-installer`.
```
Comment (alexandra-simeonova) [inline docs/03-02-examples.md:59]:
```suggestion
- For example, if your `{IMAGE}` is `user/my-kyma-installer:v1.4.0`, then this command will build an image from your local sources and push it your repository `user/my-kyma-installer` with the tag `v1.4.0`. Then, this image `user/my-kyma-installer:v1.4.0` will be used in the Deployment for `kyma-installer`.
```

File: docs/03-02-examples.md [hunk 4]
```diff
@@ -26,24 +26,100 @@ To install Kyma using your own domain, run:
...
+
+To install kyma with only specific components, run:
```
Comment (alexandra-simeonova) [inline docs/03-02-examples.md:61]:
```suggestion
To install Kyma with only specific components, run:
```

File: docs/03-02-examples.md [hunk 5]
```diff
@@ -26,24 +26,100 @@ To install Kyma using your own domain, run:
...
+      installer: overrides
+      component: monitoring
+      kyma-project.io/installation: ""
+  data:
+    alertmanager.alertmanagerSpec.resources.limits.memory: "304Mi"
+    alertmanager.alertmanagerSpec.resources.requests.memory: "204Mi"
+  ```
+- In this example, overrides are provided for 2 different components: `ory` and `monitoring`. For `ory`, the values of `hydra.deployment.resources.limits.cpu` and `hydra.deployment.resources.requests.cpu` will be overriden to `153m` and `53m` respectively. For `monitoring`, the values of `alertmanager.alertmanagerSpec.resources.limits.memory` and `alertmanager.alertmanagerSpec.resources.requests.memory` will be overriden to `304Mi` and `204Mi` respectively.
+
+- It is also possible to provide multiple override files at the same time
```
Comment (alexandra-simeonova) [inline docs/03-02-examples.md:118]:
```suggestion
- It is also possible to provide multiple override files at the same time:
```

**Review by suleymanakbas91 — APPROVED** [review]

**Review by suleymanakbas91 — CHANGES_REQUESTED** [review]
Whatever written in the install Kyma part is also applicable for the upgrade. How about we use install/upgrade everywhere in this part instead of only install? Does it make sense? Otherwise, we would need another part for the upgrade. @a-thaler @shorim

**Comment by a-thaler** [comment]
Yes, I'm fine with combining it.

**Review by suleymanakbas91 — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Pull request merged**